sql: Exempt SET CLUSTER SETTING from throttling

spilchen opened the pull request:
This change prevents the SET CLUSTER SETTING ... command from being throttled. Previously, it was subject to throttling, but since this command is used to install a new license and disable throttling, we want to ensure that the action itself is not throttled. This adds an exception to allow license updates.

This will be backported to versions 24.2, 24.1, 23.2, and 23.1.

Epic: CRDB-39988
Closes CRDB-42308
Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/131303)
<!-- Reviewable:end -->

fqazi reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

> Reviewed 2 of 2 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/131303#-:-O7emTA1AJVFxrCfiasU:bnfp4nl)* status: <span>:shipit:</span> complete! 1 of 0 LGTMs obtained (waiting on @spilchen)

___
*[`pkg/sql/conn_executor_exec.go` line 512 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/131303#-O7emIVf8GzS0bICft8Q:-O7emIVf8GzS0bICft8R:bi1zdd6) ([raw file](https://github.com/cockroachdb/cockroach/blob/12b33e5d2c45ce60bf09d42ee32e4cb6b6e0b7e7/pkg/sql/conn_executor_exec.go#L512)):*
> ```Go
> 		// to disable throttling. We want to avoid the situation where the action
> 		// to disable throttling is itself throttled.
> 		if _, ok := ast.(*tree.SetClusterSetting); !ok {
> ```

Nit: If we see other statements that need this bypass, we might want to have a helper function here.


<!-- Sent from Reviewable.io -->

spilchen committed:
sql: Exempt SET CLUSTER SETTING from throttling

This change prevents the SET CLUSTER SETTING ... command from being
throttled. Previously, it was subject to throttling, but since this
command is used to install a new license and disable throttling, we want
to ensure that the action itself is not throttled. This adds an
exception to allow license updates.

This will be backported to versions 24.2, 24.1, 23.2, and 23.1.

Epic: CRDB-39988
Closes CRDB-42308
Release note: None

spilchen reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/131303)* status: <span>:shipit:</span> complete! 0 of 0 LGTMs obtained (and 1 stale) (waiting on @fqazi)

___
*[`pkg/sql/conn_executor_exec.go` line 512 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/131303#-O7emIVf8GzS0bICft8Q:-O7fF3L62Ngc-JCP82Uw:bdnz4qv) ([raw file](https://github.com/cockroachdb/cockroach/blob/12b33e5d2c45ce60bf09d42ee32e4cb6b6e0b7e7/pkg/sql/conn_executor_exec.go#L512)):*
<details><summary><i>Previously, fqazi (Faizan Qazi) wrote…</i></summary><blockquote>

Nit: If we see other statements that need this bypass, we might want to have a helper function here.
</blockquote></details>

Good point! I’ve added the helper as you suggested.


<!-- Sent from Reviewable.io -->

fqazi reviewed (approved):
> Reviewed 1 of 1 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/131303)* status: <span>:shipit:</span> complete! 0 of 0 LGTMs obtained (and 1 stale) (waiting on @spilchen)



<!-- Sent from Reviewable.io -->

spilchen commented:
TFTR @fqazi 

bors r=fqazi
